sdk/storage/storage-queue/samples/typescript/iterators.ts

```diff
@@ -23,21 +23,82 @@ async function main() {
   console.log(`List queues`);
 
   // List queues
-  let iter1 = queueServiceClient.listQueues();
   let i = 1;
+  let marker: string | undefined = undefined;
+
+  let iter1 = queueServiceClient.listQueues();
   for await (const item of iter1) {
-    console.log(`Queue${i}: ${item.name}`);
+    console.log(`Queue${i}: ${item!.name}`);
+    i++;
+  }
+
+  // Same as the previous example
+  i = 1;
+  for await (const item of queueServiceClient.listQueues()) {
+    console.log(`Queue${i}: ${item!.name}`);
+    i++;
+  }
+
+  // Generator syntax .next()
+  i = 1;
+  let iter2 = queueServiceClient.listQueues();
+  let item2 = (await iter2.next()).value;
+  do {
+    console.log(`Queue${i}: ${item2!.name}`);
     i++;
+    item2 = (await iter2.next()).value;
+  } while (item2);
+
+  ////////////////////////////////
+  ///  Examples for .byPage()  ///
+  ////////////////////////////////
+
+  i = 1;
+  for await (const item1 of queueServiceClient.listQueues().byPage({})) {
+    item1.queueItems!.forEach((queueItem) => {
+      console.log(`Queue${i}: ${queueItem.name}`);
+      i++;
+    });
+  }
+
+  // Same as the previous example
+  i = 1;
+  for await (const item2 of queueServiceClient.listQueues().byPage({ pagesize: 20 })) {
+    item2.queueItems!.forEach((queueItem) => {
+      console.log(`Queue${i}: ${queueItem.name}`);
+      i++;
+    });
   }
 
-  // List queues - generator syntax
-  let iter2 = await queueServiceClient.listQueues();
+  // Generator syntax .next()
   i = 1;
-  let item = await iter2.next();
+  let iter3 = queueServiceClient.listQueues().byPage({ pagesize: 2 });
+  let item3 = (await iter3.next()).value;
   do {
-    console.log(`Queue${i++}: ${item.value.name}`);
-    item = await iter2.next();
-  } while (item.value);
+    for (const queueItem of item3.queueItems!) {
+      console.log(`Queue${i}: ${queueItem.name}`);
+      i++;
+    }
+    item3 = (await iter3.next()).value;
+  } while (item3);
+
+  // Passing marker as an argument (similar to the previous example)
+  i = 1;
+  let iter4 = queueServiceClient.listQueues().byPage({ pagesize: 2 });
+  let item = (await iter4.next()).value;
+  // Prints 2 queuenames
+  for (const queueItem of item.queueItems!) {
+    console.log(`Queue${i}: ${queueItem.name}`);
+    i++;
+  }
+  marker = item.nextMarker;
```

Comment:
nit: `marker` is first used here.  maybe declare here instead? `let marker: string | undefined = item.nextMarker;`?